addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1730,7 +1738,18 @@ const RangeUserValueWidget = UnitUserValueWidget.extend({
      */
     async setValue(value, methodName) {
         await this._super(...arguments);
-        this.input.value = this._value;
+        const customValues = this._methodsParams.optionsPossibleValues[methodName];
+        this.input.value = this.isCustomRange ? customValues.indexOf(value) : this._value;
+    },
+    /**
+     * @override
+     */
+    getValue(methodName) {
+        const value = this._super(...arguments);
+        if (this.isCustomRange) {
+            return this._methodsParams.optionsPossibleValues[methodName][+value];
+        }
+        return value;
```

Comment:
See here, relying on `isCustomRange` is just wrong: you check the `optionsPossibleValues` if the whole widget is considered "custom" while it is in fact per method. If the widget is declared like this:
```xml
<we-range data-select-class="a|b|c" data-set-number=""/>
```
I expect a 3 values slider which will call `selectClass` with either `"a"`, `"b"` or `"c"` *and* `setNumber` with either `0`, `1` or `2`. With your code I think `setNumber` will always receive falsy values. There are two solutions: either here the `getValue` checks if the possible values for the `methodName` are there and return the correct one, otherwise return the number. Or, you override `loadsMethodData` so that each method have `optionsPossibleValues` with numbers if nothing custom is there. Whatever works, the first one is probably the easiest.